app/src/components/SettingsUsers.tsx

```
@@ -1,86 +1,104 @@
 import {
-    IonButton,
-    IonItem,
-    IonModal,
-    IonList,
-  } from '@ionic/react';
+  IonButton,
+  IonAlert,
+  IonInput,
+  IonRow
+} from '@ionic/react';
 import React, { useState, useEffect } from 'react';
 import SettingsUser from '../components/SettingsUser';
 import { projectServices } from '../services/ProjectServices'
 import { userService } from '../services/UserServices'
 
 interface ContainerProps {
-    project: string;
-    firebase:any
+  project: string;
+  firebase: any
 }
 
 const SettingsUsers: React.FC<ContainerProps> = ({ project, firebase }) => {
-    const [showUserModal, setShowUserModal] = useState(false);
+  const [showAlert, setShowAlert] = useState(false);
+  const [errorMessage, setErrorMessage] = useState<string>();
+  const [newUser, setNewUser] = useState<string>();
 
-    const initialUsers = [
-        { id: 0, email: 'No users'}
-    ]
+  const initialUsers = [
+    { id: 0, email: 'No users' }
+  ]
 
-    const [users, setUsers] = useState(initialUsers);
-    useEffect(() => {
-      try {
-        console.log("firebase " + firebase.auth.currentUser)
-        projectServices.getProjectUsers(project, firebase)
-        // userService.getAllUsers() // testing only
+  const [users, setUsers] = useState(initialUsers);
+  useEffect(() => {
+    try {
+      console.log("firebase " + firebase.auth.currentUser)
+      projectServices.getProjectUsers(project, firebase)
         .then(data => {
           setUsers(data)
         })
-      } catch (e) {
-        
-      }
-    }, [])
+    } catch (e) {
 
-    const [allUsers, setAllUsers] = useState(initialUsers);
-    useEffect(() => {
-      try {
-        userService.getAllUsers()
+    }
+  }, [])
+
+  const [allUsers, setAllUsers] = useState(initialUsers);
+  useEffect(() => {
+    try {
+      userService.getAllUsers()
```

the service getAllUsers() is old and now requires a page and page_size for backend API as a query.
And this causes a problem when trying to check if the given email exists in the database since you can't fetch all the users at once now. It's likely that there will be a lot of users in the future and getting too much data isn't good. 

So I think we need to set up the backend API that checks through the users with the given email and add by the email. If there is no user with given email, it should return an error, so then we can catch that error to say on the modal

---

I agree, we can make an API that requires the email to be sent in the request body and behaves as In Ha has described above.

We would also need to make a new service for this API.

---

PR for API and service can be found here: https://github.com/saddboys/Collaborative_content_coding/pull/74

---

updated to not call getAllUsers()